### PR TITLE
integrate npe2 schema updates 

### PR DESCRIPTION
**Pull request by nclack**
# Description

This integrates changes from a final pass of npe2 schema changes. See napari/npe2#49.

Documentation updates will be made in #3769 

# How has this been tested?
The test suite was run with and without the presence of npe2.
- [x] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: napari/components/_tests/test_layers_list.py [hunk 1]
```diff
@@ -6,6 +6,15 @@
...
 
+try:
+    import npe2  # noqa: F401
+
+    BUILTINS = 'napari'
+    SVG = 'napari-svg'
+except ImportError:
+    BUILTINS = 'builtins'
+    SVG = 'svg'
+
```
Comment (nclack) [inline napari/components/_tests/test_layers_list.py:17]: in npe2, the plugin name should be the python package name. A consequence is that the npe2 versions of these plugins has a different name. Tests that rely on plugin lookup by name need the new name. Hence this.

See napari/npe2#57 for a proposed mechanism to let plugins keep their old names for the purposes of these lookups.

**Comment by nclack** [comment]
I don't know what's going on with the test failures here.  They look unrelated.

**Comment by tlambert03** [comment]
looks like there was a dask release yesterday, that is causing an additional compute from what we expect

**Review by tlambert03 — APPROVED** [review]
LGTM if this is ready to go?

**Pull request merged**